lib/trino-orc/src/test/java/io/trino/orc/OrcTester.java

```diff
@@ -971,16 +971,26 @@ public static DataSize writeOrcFileColumnHive(File outputFile, RecordWriter reco
             throws Exception
     {
         SettableStructObjectInspector objectInspector = createSettableStructObjectInspector("test", type);
+        return writeOrcFileColumnHive(outputFile, recordWriter, objectInspector,  ImmutableList.of(type), ImmutableList.of(values));
+    }
+
+    public static DataSize writeOrcFileColumnHive(File outputFile, RecordWriter recordWriter, SettableStructObjectInspector objectInspector, List<Type> types, List<Iterator<?>> data)
```

Comment:
This public method signature is too brittle at the moment.

`List<Iterator<?>> data` can be composed out of iterators of different length.


One option: What about grouping: `type` and his associated `data`/ `values` under a java `record`?

There is very likely a more elegant / less brittle way to expose the functionality of this method.

Reply:
outdated